src/features/RenameSymbol.ts

```
@@ -0,0 +1,132 @@
+            const response = await client.sendRequest(RenameSymbolRequestType, req);
+
+            if (!response.changes.length) {
+                return undefined;
+            }
+
+            const edit = new WorkspaceEdit();
+            response.changes.forEach(change => {
+                const uri = Uri.file(change.fileName);
+
+                change.changes.forEach(change => {
+                    edit.replace(uri,
+                        new Range(change.startLine, change.startColumn, change.endLine, change.endColumn),
+                        change.newText);
+                });
+            });
+            return edit;
+        }catch (error) {
+            return undefined;
+        }
+    }
+    public async prepareRename(document: vscode.TextDocument, position: vscode.Position, _token: vscode.CancellationToken): Promise<vscode.Range | {
+        range: vscode.Range; placeholder: string;} | null> {
+
+        const req:IRenameSymbolRequestArguments = {
+            FileName : document.fileName,
+            Line: position.line,
+            Column : position.character,
+        };
+
+        try {
+            const client = await LanguageClientConsumer.getLanguageClient();
+            const response = await client.sendRequest(PrepareRenameSymbolRequestType, req);
+
+            if (!response.message) {
+                return null;
+            }
+            const wordRange = document.getWordRangeAtPosition(position);
+            if (!wordRange) {
+                return Promise.reject("Not a valid location for renaming.");
+
+            }
+            const wordText = document.getText(wordRange);
+            if (response.message) {
+                return Promise.reject(response.message);
+
+            }
+
+            return {
+                range: wordRange,
+                placeholder: wordText
+            };
+        }catch (error) {
+            return null;
```

Probably shouldn't swallow errors, these should at least be surfaced in the logs if not returned to the user outright in an errormessage

---

Or a promise rejection

---

Fixed, thanks!